### PR TITLE
feat(builtin): add vault service

### PR DESCRIPTION
**Pull request by justprosh**
depends on fluencelabs/nox#1631

### CODE REVIEW

File: builtin.aqua [hunk 1]
```diff
@@ -381,3 +382,10 @@ service Debug("debug"):
 
 service Console("run-console"):
     print(any: ⊤)
+
+-- Particle File Vault
+service Vault("vault"):
+    -- Puts data into the vault and returns virtual path
+    put(data: string) -> Path
+    -- Cats data from the vault
```
Comment (kmd-fl) [inline builtin.aqua:390]:
```suggestion
    -- Returns the content of the file from the vault
```
mb?

File: builtin.aqua [hunk 2]
```diff
@@ -381,3 +382,10 @@ service Debug("debug"):
...
+    cat(path: Path) -> string
```
Comment (kmd-fl) [inline builtin.aqua:391]: Come to think of it, isn't it also possible for the file to be binary? For example, when reading from IPFS. What will happen then? I guess I should've asked in the other PR.

Comment (justprosh) [reply builtin.aqua:391]: That is a good point, but I don't know any use case for it. With string, you can do a lot with existent builtins, but to do smth with binary, you should have another service that can use vault.

Comment (kmd-fl) [reply builtin.aqua:391]: aqua-ipfs is reading to the vault, no? So it should be easy to do smth like
```
result <- Ipfs.get(CID)
??? <- Vault.cat(result.path)
```

Comment (justprosh) [reply builtin.aqua:391]: We have Ipfs.cat

**Pull request merged**